cmd/fluxctl/root_cmd.go

```diff
@@ -74,11 +82,38 @@ func (opts *rootOpts) Command() *cobra.Command {
 }
 
 func (opts *rootOpts) PersistentPreRunE(cmd *cobra.Command, _ []string) error {
+	opts.Namespace = getFromEnvIfNotSet(cmd.Flags(), "k8s-forward-namespace", opts.Namespace, envVariableNamespace)
```

Comment:
There's a difference between the default namespace (which is determined by your kubeconfig), and the namespace `"default"`. Some people use contexts in kubeconfig -- or different kubeconfigs entirely -- when operating in different namespaces. Since the port forwarding library is implicitly getting things from a kubeconfig anyway, to be able to connect at all, could it also pick up the default namespace?

Reply:
Good point! I'm still working on trying to figure out how to fetch the default namespace from the kube config (it's conspiciously missing from https://godoc.org/k8s.io/client-go/rest#Config).

Reply:
I think https://godoc.org/k8s.io/client-go/tools/clientcmd has the bits needed. `k8s.io/client-go` really is labyrinthine isn't it :-/